engine/src/main/java/org/hibernate/search/query/dsl/sort/SortNativeContext.java

```diff
@@ -0,0 +1,13 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.query.dsl.sort;
+
+/**
+ * @author Emmanuel Bernard emmanuel@hibernate.org
+ */
+public interface SortNativeContext extends SortAdditionalSortFieldContext, SortTermination {
```

Comment:
In the 5.x series setSort cannot be deprecated. The DSL output is what is passed to setSort. In 6 all bets are off but I am not sure how to do it. 

I personally like how one can mix normal and native declarations. But we can remove it if people don't agree with me.